subxt/src/storage/storage_key.rs

```diff
@@ -0,0 +1,445 @@
+use crate::{
+    error::{Error, MetadataError, StorageAddressError},
+    utils::{Encoded, Static},
+};
+use scale_decode::{visitor::IgnoreVisitor, DecodeAsType};
+use scale_encode::EncodeAsType;
+use scale_info::{PortableRegistry, TypeDef};
+use scale_value::Value;
+use subxt_metadata::{StorageEntryType, StorageHasher};
+
+use derivative::Derivative;
+
+use super::utils::hash_bytes;
+
+#[derive(Debug, Clone)]
+// An iterator over all type ids of the key and the respective hashers.
+pub struct StorageHashersIter {
+    hashers_and_ty_ids: Vec<(StorageHasher, u32)>,
+    idx: usize,
+}
+
+impl StorageHashersIter {
+    /// Creates a new [`StorageHashersIter`]. Looks at the [`StorageEntryType`] and
+    /// assigns a hasher to each type id that makes up the key.
+    pub fn new(storage_entry: &StorageEntryType, types: &PortableRegistry) -> Result<Self, Error> {
```

Comment:
Can this be `pub(crate)` so that constructing this is an internal-to-subxt thing only? :)